### PR TITLE
FACES-2697 Fix portal:navbarSearch in Liferay 7 (reduce JSP API)

### PR DESCRIPTION
**Pull request by stiemannkj1**

### CODE REVIEW

**Comment by juangon** [comment]
Thanks kyle! I like your changes. Seems license changed in this file, can you fix that?

File: src/main/java/com/liferay/faces/util/jsp/internal/PageContextStringImpl.java [hunk 1]
```diff
@@ -314,6 +314,11 @@ public void initialize(Servlet servlet, ServletRequest request, ServletResponse
 	}
 
 	@Override
+	public BodyContent pushBody() {
+		return new BodyContentStringImpl(new JspWriterStringImpl());
```
Comment (stiemannkj1) [inline src/main/java/com/liferay/faces/util/jsp/internal/PageContextStringImpl.java]: @ngriffin7a, [this method returns `null` by default](http://grepcode.com/file/repo1.maven.org/maven2/javax.servlet/servlet-api/2.3/javax/servlet/jsp/PageContext.java#PageContext.pushBody%28%29). So my changes shouldn't break anything. I also tested `portal:navbarSearch` and `portal:inputRichText` and they worked correctly.

**Comment by stiemannkj1** [comment]
Thanks @juangon, the license in Util is Apache (https://github.com/liferay/liferay-faces-util/blob/master/src/main/java/com/liferay/faces/util/client/BrowserSniffer.java#L1-L15), but since `BodyContentStringImpl.java` was copied from Bridge Ext, it still had the LGPL. Jalopy fixed it automatically for me.

**Comment by juangon** [comment]
Ooops didn't know that was already wrong. Tomorrow will make some checks but everything seems fine.
Thanks!

**Comment by ngriffin7a** [comment]
Merged and backported to 2.x. Thank you.

**Pull request merged**